prov/gni/test/cq.c

```diff
@@ -212,6 +213,7 @@ void cq_teardown(void)
 {
 	cr_assert(!fi_close(&rcq->fid), "failure in closing cq.");
 	teardown();
+	mode_bits = old_mode_bits;
```

Comment:
This function gets called with `setup` which doesn't set old_mode_bits.  I don't think this resetting works in that case.